### PR TITLE
Adding Python Docstring

### PR DESCRIPTION
**Pull request by SeqLaz**

### CODE REVIEW

File: qfieldcloud_sdk/sdk.py [hunk 1]
```diff
@@ -58,11 +92,30 @@ class ProjectCollaboratorRole(str, Enum):
 
 
 class OrganizationMemberRole(str, Enum):
+    """Defines roles for organization members.
+
+    Attributes:
+        ADMIN (str): Administrator role.
+        MEMBER (str): Member role.
+    """
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:100]:
```suggestion
    """Defines roles for organization members.
    
    Read more in the documentation: https://docs.qfield.org/reference/qfieldcloud/permissions/#roles_2

    Attributes:
        ADMIN (str): Administrator role.
        MEMBER (str): Member role.
    """
```

File: qfieldcloud_sdk/sdk.py [hunk 2]
```diff
@@ -73,41 +126,91 @@ class CollaboratorModel(TypedDict):
...
+
+    Todo:
+        * Future work that can be surely expected, check QF-4535
+    """
+
     member: str
     role: OrganizationMemberRole
     organization: str
     is_public: bool
-    # TODO future work that can be surely expected, check QF-4535
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:80]: Do not remove this please

File: qfieldcloud_sdk/sdk.py [hunk 3]
```diff
@@ -73,41 +126,91 @@ class CollaboratorModel(TypedDict):
...
+        token (str): The authentication token for API access.
+        verify_ssl (bool): Whether to verify SSL certificates.
+    """
+
     def __init__(
         self, url: str = None, verify_ssl: bool = None, token: str = None
     ) -> None:
         """Prepares a new client.
 
+        If the `verify_ssl` is set to `False`, SSL warnings will be disabled. The session is configured with retries for GET requests on specific 5xx HTTP status codes.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:201]:
```suggestion
        The session is configured with retries for GET requests on specific 5xx HTTP status codes.
```

File: qfieldcloud_sdk/sdk.py [hunk 4]
```diff
@@ -138,6 +241,13 @@ def login(self, username: str, password: str) -> Dict:
         Args:
             username: the username or the email used to register
             password: the password associated with that username
+
+        Returns:
+            Dict: A dictionary containing the authentication token and additional metadata.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:246]:
```suggestion
            Dict[str, Any]: Authentication token and additional metadata.
```

Can you please add the key and value types in all occurances for Dict and List for the code related to this PR?

File: qfieldcloud_sdk/sdk.py [hunk 5]
```diff
@@ -156,7 +266,14 @@ def login(self, username: str, password: str) -> Dict:
         return payload
 
     def logout(self) -> None:
-        """Logout from the current session."""
+        """Logout from the current session.
+
+        Returns:
+            None
+
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:273]: No need for this.

File: qfieldcloud_sdk/sdk.py [hunk 6]
```diff
@@ -183,14 +310,28 @@ def list_projects(
...
         self, project_id: str, skip_metadata: bool = True
     ) -> List[Dict[str, Any]]:
+        """Lists the files available in the specified project.
+
+        Args:
+            project_id (str): The ID of the project to list files for.
+            skip_metadata (bool): Whether to skip fetching metadata for the files. Defaults to True.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries containing file details.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:320]:
```suggestion
            List[Dict[str, Any]]: A list of file details.
```

File: qfieldcloud_sdk/sdk.py [hunk 7]
```diff
@@ -30,26 +30,60 @@
...
     FAILED = "FAILED"
 
 
 class FileTransferType(Enum):
+    """Represents the type of file transfer.
+
+    Attributes:
+        PROJECT (str): Refers to a project file.
+        PACKAGE (str): Refers to a package Type.
+    """
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:56]:
```suggestion
    """Represents the type of file transfer.
    
    The PACKAGE transfer type is used only internally in QFieldCloud workers, so it should never be used by other API clients.

    Attributes:
        PROJECT (str): Refers to a project file.
        PACKAGE (str): Refers to a project file within the project package.
    """
```

File: qfieldcloud_sdk/sdk.py [hunk 8]
```diff
@@ -30,26 +30,60 @@
...
+
     PACKAGE = "package"
     APPLY_DELTAS = "delta_apply"
     PROCESS_PROJECTFILE = "process_projectfile"
 
 
 class ProjectCollaboratorRole(str, Enum):
+    """Defines roles for project collaborators.
+
+    Attributes:
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:79]:
```suggestion
    """Defines roles for project collaborators.

    See project collaborator roles documentation: https://docs.qfield.org/reference/qfieldcloud/permissions/#roles_1

    Attributes:
```

File: qfieldcloud_sdk/sdk.py (hunk 1 above)
Comment (SeqLaz) [reply qfieldcloud_sdk/sdk.py]: Great!

File: qfieldcloud_sdk/sdk.py [hunk 9]
```diff
@@ -183,14 +312,28 @@ def list_projects(
...
+            * Remove this temporary decoration with `etag` key
+        """
         params = {}
 
         if skip_metadata:
             params["skip_metadata"] = "1"
 
         resp = self._request("GET", f"files/{project_id}", params=params)
         remote_files = resp.json()
-        # TODO remove this temporary decoration with `etag` key
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:193]: Do not remove from here

File: qfieldcloud_sdk/sdk.py [hunk 10]
```diff
@@ -183,14 +312,28 @@ def list_projects(
...
+            skip_metadata (bool): Whether to skip fetching metadata for the files. Defaults to True.
+
+        Returns:
+            List[Dict[str, Any]]: A list of file details.
+
+        Example:
+            client.list_remote_files("project_id", True)
+
+        Todo:
+            * Remove this temporary decoration with `etag` key
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:328]: Do not keep this one here, it should be in the context where it comes from. This TODO is more for API or functionality that affects the public API.

File: qfieldcloud_sdk/sdk.py [hunk 11]
```diff
@@ -202,6 +345,20 @@ def create_project(
         description: str = "",
         is_public: bool = False,
     ) -> Dict:
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:347]:
```suggestion
    ) -> Dict[str, Any]:
```

**Review by suricactus — COMMENTED** [review]
Please go over the whole docs once again, as the coments I made were not always properly aligned. Please squash and force push a single commit.

File: qfieldcloud_sdk/sdk.py (hunk 10 above)
Comment (SeqLaz) [reply qfieldcloud_sdk/sdk.py:328]: Great!.... I'll keep on mind that

File: qfieldcloud_sdk/sdk.py [hunk 12]
```diff
@@ -30,26 +30,64 @@
...
+        SUCCESS (str): The transfer was successful.
+        FAILED (str): The transfer failed.
+    """
+
     PENDING = "PENDING"
     SUCCESS = "SUCCESS"
     FAILED = "FAILED"
 
 
 class FileTransferType(Enum):
```
Comment (nirvn) [inline qfieldcloud_sdk/sdk.py:50]: Should this also not be a str, Enum?
```suggestion
class FileTransferType(str, Enum):
```

File: qfieldcloud_sdk/sdk.py [hunk 13]
```diff
@@ -85,29 +153,64 @@ class OrganizationMemberModel(TypedDict):
...
+
     limit = None
     offset = None
 
     def __init__(
         self, limit: Optional[int] = None, offset: Optional[int] = None
     ) -> None:
+        """Initializes the pagination settings.
+
+        Args:
```
Comment (nirvn) [inline qfieldcloud_sdk/sdk.py:171]: Should we use the non abbreviated keyword here (we don't use abbreviations anywhere else).

File: qfieldcloud_sdk/sdk.py [hunk 14]
```diff
@@ -85,29 +153,64 @@ class OrganizationMemberModel(TypedDict):
...
+    """The core component of the QFieldCloud SDK, providing methods for interacting with the QFieldCloud platform.
+
+    This class handles authentication, project management, file management, and more.
+
+    Attributes:
+        session (requests.Session): The session object to maintain connections.
+        url (str): The base URL for the QFieldCloud API.
+        token (str): The authentication token for API access.
+        verify_ssl (bool): Whether to verify SSL certificates.
+    """
```
Comment (nirvn) [inline qfieldcloud_sdk/sdk.py:198]: Do we want to come up with a rule for class attributes (i.e. should they all be declared above __init__ with default values? Less risks of creating situations where an attribute isn't initialized in __init__ and cause an error?

File: qfieldcloud_sdk/sdk.py (hunk 12 above)
Comment (suricactus) [reply qfieldcloud_sdk/sdk.py:50]: Good spot! Ideally should be https://docs.python.org/3/library/enum.html#enum.StrEnum, but it's 3.11 onwards (I am looking for bigger backwards compatibility for this kind of libs).

File: qfieldcloud_sdk/sdk.py (hunk 13 above)
Comment (suricactus) [reply qfieldcloud_sdk/sdk.py:171]: If you refer to Args, it is part of https://google.github.io/styleguide/pyguide.html "specification", which is widely used in different documentation setups.

File: qfieldcloud_sdk/sdk.py (hunk 14 above)
Comment (suricactus) [reply qfieldcloud_sdk/sdk.py:198]: The thing is that sometimes these attributes are not nullable but also cannot have defaults that make sense. What I would suggest is we added them below like this:

```suggestion
    Attributes:
        session (requests.Session): The session object to maintain connections.
        url (str): The base URL for the QFieldCloud API.
        token (str): The authentication token for API access.
        verify_ssl (bool): Whether to verify SSL certificates.
    """

    session: requests.Session

    url: str

    token:str

    veryfy_ssl: bool
```

File: qfieldcloud_sdk/sdk.py [hunk 15]
```diff
@@ -85,29 +153,64 @@ class OrganizationMemberModel(TypedDict):
...
+
     def __init__(
         self, url: str = None, verify_ssl: bool = None, token: str = None
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:201]:
```suggestion
        self,
        url: str = "",
        verify_ssl: bool = True,
        token: str = "",
```

Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:201]:
```suggestion
        self,
        url: str = "",
        verify_ssl: bool = True,
        token: str = "",
```

File: qfieldcloud_sdk/sdk.py [hunk 16]
```diff
@@ -85,29 +153,64 @@ class OrganizationMemberModel(TypedDict):
...
 
-        If the `url` is not provided, uses `QFIELDCLOUD_URL` from the environment.
-        If the `token` is not provided, uses `QFIELDCLOUD_TOKEN` from the environment.
-        `session` will be reused between requests if the SDK is run as a library.
+        The session is configured with retries for GET requests on specific 5xx HTTP status codes.
+
+        Args:
+            url (Optional[str]): The base URL for the QFieldCloud API. Defaults to `QFIELDCLOUD_URL` environment variable if not provided.
+            verify_ssl (Optional[bool]): Whether to verify SSL certificates. Defaults to True if not specified.
+            token (Optional[str]): The authentication token for API access. Defaults to `QFIELDCLOUD_TOKEN` environment variable if not provided.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:210]:
```suggestion
            url (str): The base URL for the QFieldCloud API. Defaults to `QFIELDCLOUD_URL` environment variable if empty.
            verify_ssl (bool): Whether to verify SSL certificates. Defaults to True.
            token (str): The authentication token for API access. Defaults to `QFIELDCLOUD_TOKEN` environment variable if empty.
```

Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:210]:
```suggestion
            url (str): The base URL for the QFieldCloud API. Defaults to `QFIELDCLOUD_URL` environment variable if empty.
            verify_ssl (bool): Whether to verify SSL certificates. Defaults to True.
            token (str): The authentication token for API access. Defaults to `QFIELDCLOUD_TOKEN` environment variable if empty.
```

File: qfieldcloud_sdk/sdk.py [hunk 17]
```diff
@@ -132,12 +235,19 @@ def __init__(
...
+        """Logs in with the provided username and password.
 
         Args:
-            username: the username or the email used to register
-            password: the password associated with that username
+            username (str): The username or email used to register.
+            password (str): The password associated with the username.
+
+        Returns:
+            Dict[str, Any]: Authentication token and additional metadata.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:246]:
```suggestion
            dict[str, Any]: Authentication token and additional metadata.
```

File: qfieldcloud_sdk/sdk.py [hunk 18]
```diff
@@ -201,7 +332,18 @@ def create_project(
...
         is_public: bool = False,
-    ) -> Dict:
+    ) -> Dict[str, Any]:
+        """Creates a new project in QFieldCloud.
+
+        Args:
+            name (str): The name of the new project.
+            owner (Optional[str]): The owner of the project. Defaults to None.
+            description (Optional[str]): A description of the project. Defaults to an empty string.
+            is_public (Optional[bool]): Whether the project should be public. Defaults to False.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:342]:
```suggestion
            owner (str, optional): The owner of the project. Defaults to None.
            description (str, optional): A description of the project. Defaults to an empty string.
            is_public (bool, optional): Whether the project should be public. Defaults to False.
```

File: qfieldcloud_sdk/sdk.py [hunk 19]
```diff
@@ -201,7 +332,18 @@ def create_project(
...
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the details of the created project.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:345]:
```suggestion
            dict[str, Any]: A dictionary containing the details of the created project.
```

File: qfieldcloud_sdk/sdk.py [hunk 20]
```diff
@@ -167,9 +281,14 @@ def list_projects(
...
         **kwargs,
     ) -> List[Dict[str, Any]]:
-        """
-        Returns a list of projects accessible to the current user,
-        their own and optionally the public ones.
+        """Returns a list of projects accessible to the current user, their own and optionally the public ones.
+
+        Args:
+            include_public (Optional[bool]): Whether to include public projects in the list. Defaults to False.
+            pagination (Pagination): Pagination settings for the request. Defaults to an empty Pagination instance.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:288]:
```suggestion
            include_public (bool, optional): Whether to include public projects in the list. Defaults to False.
            pagination (Pagination, optional): Pagination settings for the request. Defaults to an empty Pagination instance.
```

File: qfieldcloud_sdk/sdk.py [hunk 21]
```diff
@@ -167,9 +281,14 @@ def list_projects(
         pagination: Pagination = Pagination(),
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:281]: The line before is wrongly marked as `Optional`, please remove.

File: qfieldcloud_sdk/sdk.py [hunk 22]
```diff
@@ -85,29 +153,75 @@ class OrganizationMemberModel(TypedDict):
...
     offset = None
 
     def __init__(
         self, limit: Optional[int] = None, offset: Optional[int] = None
     ) -> None:
+        """Initializes the pagination settings.
+
+        Args:
+            limit (Optional[int]): The maximum number of items to return. Defaults to None.
+            offset (Optional[int]): The starting point from which to return items. Defaults to None.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:173]:
```suggestion
            limit (int | None, optional): The maximum number of items to return. Defaults to None.
            offset (int | None, optional): The starting point from which to return items. Defaults to None.
```

File: qfieldcloud_sdk/sdk.py [hunk 23]
```diff
@@ -85,29 +153,75 @@ class OrganizationMemberModel(TypedDict):
 
 
 class Pagination:
+    """The Pagination class allows for controlling and managing pagination of results within the QFieldCloud SDK.
+
+    Attributes:
+        limit (Optional[int]): The maximum number of items to return.
+        offset (Optional[int]): The starting point from which to return items.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:160]:
```suggestion
        limit (int | None): The maximum number of items to return.
        offset (int | None): The starting point from which to return items.
```

File: qfieldcloud_sdk/sdk.py [hunk 24]
```diff
@@ -201,7 +343,18 @@ def create_project(
         owner: str = None,
         description: str = "",
         is_public: bool = False,
-    ) -> Dict:
+    ) -> Dict[str, Any]:
+        """Creates a new project in QFieldCloud.
+
+        Args:
+            name (str): The name of the new project.
+            owner (str, optional): The owner of the project. Defaults to None.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:351]:
```suggestion
            owner (str | None, optional): The owner of the project. Defaults to None.
```

File: qfieldcloud_sdk/sdk.py [hunk 25]
```diff
@@ -156,20 +277,29 @@ def login(self, username: str, password: str) -> Dict:
...
-        Returns a list of projects accessible to the current user,
-        their own and optionally the public ones.
+        """Returns a list of projects accessible to the current user, their own and optionally the public ones.
+
+        Args:
+            include_public (bool): Whether to include public projects in the list. Defaults to False.
+            pagination (Pagination): Pagination settings for the request. Defaults to an empty Pagination instance.
+
+        Returns:
+            list[Dict[str, Any]]: A list of dictionaries containing project details.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:302]:
```suggestion
        Args:
            include_public (bool, optional): Whether to include public projects in the list. Defaults to False.
            pagination (Pagination, optional): Pagination settings for the request. Defaults to an empty Pagination instance.

        Returns:
            list[dict[str, Any]]: A list of dictionaries containing project details.
```

File: qfieldcloud_sdk/sdk.py [hunk 26]
```diff
@@ -183,6 +313,18 @@ def list_projects(
...
         self, project_id: str, skip_metadata: bool = True
     ) -> List[Dict[str, Any]]:
+        """Lists the files available in the specified project.
+
+        Args:
+            project_id (str): The ID of the project to list files for.
+            skip_metadata (bool): Whether to skip fetching metadata for the files. Defaults to True.
+
+        Returns:
+            List[Dict[str, Any]]: A list of file details.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:323]:
```suggestion
        Args:
            project_id (str): The ID of the project to list files for.
            skip_metadata (bool, optional): Whether to skip fetching metadata for the files. Defaults to True.

        Returns:
            list[dict[str, Any]]: A list of file details.
```

File: qfieldcloud_sdk/sdk.py [hunk 27]
```diff
@@ -183,6 +313,18 @@ def list_projects(
...
+
+        Example:
+            client.list_remote_files("project_id", True)
+        """
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:327]:
```suggestion
        """List project files.

        Args:
            project_id (str): The ID of the project to list files for.
            skip_metadata (bool, optional): Whether to skip fetching metadata for the files. Defaults to True.

        Returns:
            list[dict[str, Any]]: A list of file details.

        Example:
            client.list_remote_files("project_id", True)
        """
```

File: qfieldcloud_sdk/sdk.py [hunk 28]
```diff
@@ -201,7 +343,18 @@ def create_project(
         owner: str = None,
         description: str = "",
         is_public: bool = False,
-    ) -> Dict:
+    ) -> Dict[str, Any]:
+        """Creates a new project in QFieldCloud.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:347]:
```suggestion
        """Creates a new project.
```

File: qfieldcloud_sdk/sdk.py [hunk 29]
```diff
@@ -216,6 +369,14 @@ def create_project(
         return resp.json()
 
     def delete_project(self, project_id: str):
+        """Delete a QFieldCloud project.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:372]:
```suggestion
        """Deletes a project.
```

File: qfieldcloud_sdk/sdk.py [hunk 30]
```diff
@@ -216,6 +369,14 @@ def create_project(
...
+
+        Args:
+            project_id (str): The ID of the project to delete.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:375]:
```suggestion
            project_id (str): Project ID.
```

File: qfieldcloud_sdk/sdk.py [hunk 31]
```diff
@@ -231,7 +392,21 @@ def upload_files(
         force: bool = False,
         job_id: str = "",
     ) -> List[Dict]:
-        """Upload files to a QFieldCloud project"""
+        """Upload files to a QFieldCloud project.
+
+        Args:
+            project_id (str): The ID of the project to upload files to.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:398]:
```suggestion
            project_id (str): Project ID.
```

And so on, it's not needed to be very verbose about this.

File: qfieldcloud_sdk/sdk.py [hunk 32]
```diff
@@ -231,7 +392,21 @@ def upload_files(
...
+            upload_type (FileTransferType): The type of file transfer (PROJECT or PACKAGE).
+            project_path (str): The local directory containing the files to upload.
+            filter_glob (str): A glob pattern to filter which files to upload.
+            throw_on_error (bool, optional): Whether to raise an error if a file fails to upload. Defaults to False.
+            show_progress (bool, optional): Whether to display a progress bar during upload. Defaults to False.
+            force (bool, optional): Whether to force upload all files, even if they exist remotely. Defaults to False.
+            job_id (str, optional): The job ID, required if `upload_type` is PACKAGE. Defaults to an empty string.
+
+        Returns:
+            List[Dict]: A list of dictionaries with information about the uploaded files.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:408]:
```suggestion
            list[dict]: A list of dictionaries with information about the uploaded files.
```

File: qfieldcloud_sdk/sdk.py [hunk 33]
```diff
@@ -298,6 +473,19 @@ def upload_file(
         show_progress: bool,
         job_id: str = "",
     ) -> requests.Response:
+        """Upload a single file to a QFieldCloud project.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:476]:
```suggestion
        """Upload a single file to a project.
```

No need to specify "QFieldCloud Project", there are no other projects in this context.

File: qfieldcloud_sdk/sdk.py [hunk 34]
```diff
@@ -298,6 +473,19 @@ def upload_file(
...
+
+        Args:
+            project_id (str): The ID of the project to upload the file to.
+            upload_type (FileTransferType): The type of file transfer (PROJECT or PACKAGE).
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:480]:
```suggestion
            upload_type (FileTransferType): The type of file transfer.
```

File: qfieldcloud_sdk/sdk.py [hunk 35]
```diff
@@ -349,8 +537,9 @@ def download_project(
             local_dir: destination directory where the files will be downloaded
             filter_glob: if specified, download only the files which match the glob, otherwise download all
             force_download (bool, optional): Download file even if it already exists locally. Defaults to False.
+        Returns:
+                List[Dict]: A list of dictionaries with information about the downloaded files.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:541]:
```suggestion
                list[dict]: A list of dictionaries with information about the downloaded files.
```

File: qfieldcloud_sdk/sdk.py [hunk 36]
```diff
@@ -370,8 +559,15 @@ def list_jobs(
...
-        Returns a paginated lists of jobs accessible to the user.
+        """Return a paginated list of jobs accessible to the user.
+
+        Args:
+            project_id (str): The ID of the project.
+            job_type (JobTypes, optional): The type of job to filter by. Defaults to None.
+            pagination (Pagination, optional): Pagination settings. Defaults to a new Pagination object.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries representing the jobs.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:570]:
```suggestion
            list[dict[str, Any]]: A list of dictionaries representing the jobs.
```

File: qfieldcloud_sdk/sdk.py [hunk 37]
```diff
@@ -370,8 +559,15 @@ def list_jobs(
         job_type: JobTypes = None,
         pagination: Pagination = Pagination(),
     ) -> List[Dict[str, Any]]:
-        """
-        Returns a paginated lists of jobs accessible to the user.
+        """Return a paginated list of jobs accessible to the user.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:562]: Everything should be paginated, no need to add this word here:
```suggestion
        """List project jobs.
```

File: qfieldcloud_sdk/sdk.py [hunk 38]
```diff
@@ -387,8 +583,16 @@ def list_jobs(
     def job_trigger(
         self, project_id: str, job_type: JobTypes, force: bool = False
     ) -> Dict[str, Any]:
-        """Initiate a new project job."""
+        """Initiate a new project job.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:586]:
```suggestion
        """Trigger a new job for given project.
```

File: qfieldcloud_sdk/sdk.py [hunk 39]
```diff
@@ -387,8 +583,16 @@ def list_jobs(
...
 
+        Args:
+            project_id (str): The ID of the project.
+            job_type (JobTypes): The type of job to trigger.
+            force (bool, optional): Whether to force the job execution. Defaults to False.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the job information.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:594]:
```suggestion
            dict[str, Any]: A dictionary containing the job information.
```

File: qfieldcloud_sdk/sdk.py [hunk 40]
```diff
@@ -402,8 +606,14 @@ def job_trigger(
...
 
     def job_status(self, job_id: str) -> Dict[str, Any]:
-        """Get job status."""
+        """Get the status of a job.
+
+        Args:
+            job_id (str): The ID of the job.
 
+        Returns:
+            Dict[str, Any]: A dictionary containing the job status.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:615]:
```suggestion
            dict[str, Any]: A dictionary containing the job status.
```

File: qfieldcloud_sdk/sdk.py [hunk 41]
```diff
@@ -517,6 +734,8 @@ def package_download(
             local_dir: destination directory where the files will be downloaded
             filter_glob: if specified, download only packaged files which match the glob, otherwise download all
             force_download (bool, optional): Download file even if it already exists locally. Defaults to False.
+        Returns:
+            List[Dict]: A list of dictionaries with information about the downloaded files.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:738]:
```suggestion
            list[dict[str, Any]]: A list of dictionaries with information about the downloaded files.
```

The dict should also be updated in the return type of the function.

File: qfieldcloud_sdk/sdk.py [hunk 42]
```diff
@@ -496,7 +706,14 @@ def delete_files(
...
 
     def package_latest(self, project_id: str) -> Dict[str, Any]:
-        """Check project packaging status."""
+        """Check the latest packaging status of a project.
+
+        Args:
+            project_id (str): The ID of the project.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the latest packaging status.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:715]:
```suggestion
            dict[str, Any]: A dictionary containing the latest packaging status.
```

File: qfieldcloud_sdk/sdk.py [hunk 43]
```diff
@@ -156,20 +277,29 @@ def login(self, username: str, password: str) -> Dict:
...
         self,
-        include_public: Optional[bool] = False,
+        include_public: bool = False,
         pagination: Pagination = Pagination(),
         **kwargs,
     ) -> List[Dict[str, Any]]:
-        """
-        Returns a list of projects accessible to the current user,
-        their own and optionally the public ones.
+        """Returns a list of projects accessible to the current user, their own and optionally the public ones.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:295]:
```suggestion
        """List projects accessible by the current user. Optionally include all public projects.
```

File: qfieldcloud_sdk/sdk.py [hunk 44]
```diff
@@ -156,20 +277,29 @@ def login(self, username: str, password: str) -> Dict:
...
+
+        Args:
+            include_public (bool): Whether to include public projects in the list. Defaults to False.
+            pagination (Pagination): Pagination settings for the request. Defaults to an empty Pagination instance.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:299]:
```suggestion
            include_public (bool, optional): Whether to include public projects in the list. Defaults to False.
            pagination (Pagination, optional): Pagination settings for the request. Defaults to an empty Pagination instance.
```

File: qfieldcloud_sdk/sdk.py [hunk 45]
```diff
@@ -85,29 +153,75 @@ class OrganizationMemberModel(TypedDict):
...
+
     def __init__(
-        self, url: str = None, verify_ssl: bool = None, token: str = None
+        self,
+        url: str = "",
+        verify_ssl: bool = True,
+        token: str = "",
     ) -> None:
-        """Prepares a new client.
+        """Initializes a new Client instance.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:214]:
```suggestion
        """Initialize a new SDK Client instance.
```

File: qfieldcloud_sdk/sdk.py [hunk 46]
```diff
@@ -85,29 +153,75 @@ class OrganizationMemberModel(TypedDict):
...
+        Args:
+            limit (int | None, optional): The maximum number of items to return. Defaults to None.
+            offset (int | None, optional): The starting point from which to return items. Defaults to None.
+        """
         self.limit = limit
         self.offset = offset
 
     @property
     def is_empty(self):
+        """Checks if both limit and offset are None, indicating no pagination settings.
```
Comment (suricactus) [inline qfieldcloud_sdk/sdk.py:180]:
```suggestion
        """Whether both limit and offset are None, indicating no pagination settings.
```
Note this is not a function, but a `@property`, which in python is  a getter that behaves as an attribute.

**Comment by suricactus** [comment]
Let's merge for now!

**Pull request merged**